WalletWasabi/Wallets/WalletFilterProcessor.cs

```diff
@@ -281,6 +282,41 @@ private async void ReorgedAsync(object? sender, FilterModel invalidFilter)
 		}
 	}
 
+	/// <summary>
+	/// Attempt to get the bitcoin block from a full node or use P2P as a fallback.
+	/// </summary>
+	private async Task<Block> KeepTryingToGetBlockAsync(uint256 blockHash, Priority priority, CancellationToken cancellationToken)
+	{
+		BlockDownloadService.IResult fullNodeResult = await BlockDownloadService.TryGetBlockAsync(TrustedFullNodeSourceRequest.Instance, blockHash, priority, cancellationToken)
```

Comment:
For me on RegTest, once we hit this line, we never ever return from it, we are waiting here indefinitely.